### PR TITLE
Enrich MCP list_connections with favorite name and path

### PR DESCRIPTION
**Pull request by neeo**
## Summary

Improves the metadata the MCP server reports for each open connection so an MCP client (Claude, Cursor, and others) can tell which saved favorite a tab is using.

## Why

Two gaps in the current `list_connections` output:

1. The reported `name` came from the document's display name. For a connection not saved to a `.spf` file that is a placeholder like `Untitled 7` (assigned by SPQueryController's untitled-document counter), so a client saw `Untitled 7` even though the window shows the real connection name.
2. There was no way for the client to know which favorite, or which favorites folder, a connection belongs to. That context is useful for multi-tab workflows and, especially in read/write mode, for confirming which environment the agent is about to act on.

## What

Each `list_connections` entry now reports:

- `name`: the connection name (`doc.name()`) - the favorite name, or `user@host` - matching what the window and tab show, instead of the untitled document name.
- `favoriteName` (new): the name of the favorite the connection was opened from.
- `favoritePath` (new): the folder path in the favorites tree, for example `Production/EU/main-db`.

`favoriteName` and `favoritePath` are omitted for ad-hoc/quick connections and for favorites that are no longer in the tree, rather than sent as empty strings (matching how `host` and `database` are already omitted when empty).

## How

- The favorite id retained on the connection (`connectionKeychainID`) is resolved against the shared favorites tree (`SPFavoritesController`). The folder path is built by walking group ancestors, excluding the synthetic tree root and the top-level "Favorites" container.
- Path assembly and id normalization are factored into a small pure `SPMCPFavorite` helper with unit tests. The tree walk runs on the main thread, consistent with the rest of the MCP data source.

## Testing

- Added `SPMCPFavoriteTests` covering path assembly, empty-group handling, and id normalization. The full MCP unit suite passes.
- Built the app. The end-to-end JSON shape is best verified by connecting via a favorite and calling `list_connections` from an MCP client.

## Context

Prompted by real-world testing where an MCP client saw a connection named "Untitled 7". This replaces that with the real connection name and adds the favorite name and path.

## Summary by CodeRabbit

* **New Features**
  * Connection details now use the connection’s configured name.
  * Cleaner connection information: empty host/database fields are omitted.
  * Connections can display favorite names and a structured favorite path when available.
* **Tests**
  * Added unit tests covering favorite path formatting and identifier normalization (including edge cases like empty/invalid values).

### CODE REVIEW

File: Source/Controllers/MCP/SPAppController+MCP.swift [hunk 1]
```diff
@@ -265,19 +265,67 @@ extension SPAppController: SPMCPDataSource {
                 guard let conn = c, conn.isConnected() else { continue }
                 var info: [String: Any] = [:]
                 info["id"] = mcpDocumentID(doc)
-                let displayName = doc.displayName()
-                info["name"] = displayName.isEmpty ? doc.host() : displayName
+                let name = doc.name()
+                info["name"] = name.isEmpty ? doc.host() : name
                 let host = doc.host()
                 if !host.isEmpty { info["host"] = host }
                 if let db = doc.database(), !db.isEmpty { info["database"] = db }
```
[the remark this answers is not in the document]
Comment (neeo) [reply Source/Controllers/MCP/SPAppController+MCP.swift:272]: Declining this one. Since the recent nullability audit, `SPDatabaseDocument.h` has `NS_ASSUME_NONNULL_BEGIN` and neither `-name` nor `-host` is marked `nullable`, so both import into Swift as non-optional `String` (only `-database` is `nullable`, which is already handled with `if let`). `-host` also coalesces internally (`if (!host) host = @"";`), so it never returns nil. `.isEmpty` is therefore safe and nothing nil-boxes into the dictionary. The `?? ""` on the referenced line 238 is now redundant post-audit; adding it here would introduce "left side of ?? has non-optional type" warnings.

File: Source/Controllers/MCP/SPMCPFavorite.swift [hunk 2]
```diff
@@ -0,0 +1,53 @@
+//
```
[the remark this answers is not in the document]
Comment (neeo) [reply Source/Controllers/MCP/SPMCPFavorite.swift:1]: Keeping the `SP` prefix here for consistency with the rest of the MCP module: `SPMCPServer`, `SPMCPHTTP`, `SPMCPJSON`, and `SPMCPReadOnlyGuard` all use `SP`, so `SPMCPFavorite` matches its siblings. I could not find the SA-prefix rule documented in the repo (checked .coderabbit.yaml, agents.md, Guides/, docs/). Happy to switch the whole MCP module to `SA` in a separate pass if the maintainers prefer that convention.

File: Source/Controllers/MCP/SPAppController+MCP.swift [hunk 3]
```diff
@@ -265,19 +265,67 @@ extension SPAppController: SPMCPDataSource {
...
+        var groups: [String] = []
+        var ancestor = node.parent as? SPTreeNode
+        while let group = ancestor, group.isGroup, group !== topContainer, group !== root {
+            if let groupName = (group.representedObject as? SPGroupNode)?.nodeName, !groupName.isEmpty {
+                groups.insert(groupName, at: 0)
+            }
+            ancestor = group.parent as? SPTreeNode
+        }
+
+        return (name, SPMCPFavorite.pathString(groups: groups, favoriteName: name))
```
[the remark this answers is not in the document]
Comment (neeo) [reply Source/Controllers/MCP/SPAppController+MCP.swift:314]: Keeping the `SP` prefix here for consistency with the rest of the MCP module: `SPMCPServer`, `SPMCPHTTP`, `SPMCPJSON`, and `SPMCPReadOnlyGuard` all use `SP`, so `SPMCPFavorite` matches its siblings. I could not find the SA-prefix rule documented in the repo (checked .coderabbit.yaml, agents.md, Guides/, docs/). Happy to switch the whole MCP module to `SA` in a separate pass if the maintainers prefer that convention.

**Comment by neeo** [comment]
@Jason-Morcos - a naming convention question for your call.

CodeRabbit flagged the new `SPMCPFavorite` type in this PR for using the `SP` prefix, citing a convention that new Swift types should use `SA`, with `SP` reserved for legacy Sequel Pro-era code. I kept `SP` here for consistency, because the rest of the MCP module already uses it: `SPMCPServer`, `SPMCPHTTP`, `SPMCPJSON`, and `SPMCPReadOnlyGuard`. Renaming only this one type to `SAMCPFavorite` would make it the lone `SA` type among `SP` siblings, which felt worse than matching the module.

We are asking because I could not find the SA-prefix rule written down anywhere in the repo (checked .coderabbit.yaml, agents.md, Guides/, and docs/), so it is not clear how strictly it should apply to a module that is already entirely `SP`. We would rather follow your intended direction than guess.

Options as I see them:
1. Keep `SP` within the MCP module for now (what this PR does), and rename the whole module `SP` -> `SA` later as part of the modernization.
2. Rename `SPMCPFavorite` -> `SAMCPFavorite` in this PR now, accepting the temporary inconsistency with its siblings.
3. Switch the entire MCP module `SP` -> `SA` in a dedicated follow-up PR (happy to open that).

Whichever you prefer - just let me know and I will adjust.

**Comment by Jason-Morcos** [comment]
Let's stick with option 1! We have a mix of SP and SA today

**Comment by Jason-Morcos** [comment]
@codex review

**Review by Jason-Morcos — APPROVED** [review]
